### PR TITLE
 add ldapsync blacklisting

### PR DESCRIPTION
**Pull request by deads2k**
still needs e2e tests

@stevekuznetsov rebase your label on this and build out unit tests for the ldap lister to ensure behavior for groups that don't have the correct annotations or only match labels, etc.

### CODE REVIEW

File: pkg/cmd/experimental/syncgroups/cli/syncgroups.go [hunk 1]
```diff
@@ -134,7 +138,9 @@ func NewCmdSyncGroups(name, fullName string, f *clientcmd.Factory, out io.Writer
 		},
 	}
 
-	cmd.Flags().StringVar(&whitelistFile, "whitelist", whitelistFile, "path to the group whitelist")
+	cmd.Flags().StringVar(&whitelistFile, "whitelist", whitelistFile, "path to the group whitelist file")
+	cmd.Flags().StringVar(&blacklistFile, "blacklist", whitelistFile, "path to the group blacklist file")
```
Comment (stevekuznetsov) [inline pkg/cmd/experimental/syncgroups/cli/syncgroups.go]: add to help text

File: pkg/cmd/experimental/syncgroups/cli/syncgroups.go [hunk 2]
```diff
@@ -134,7 +138,9 @@ func NewCmdSyncGroups(name, fullName string, f *clientcmd.Factory, out io.Writer
...
+	cmd.Flags().StringSliceVar(&options.Blacklist, "blacklist-group", options.Blacklist, "group to blacklist")
```
Comment (stevekuznetsov) [inline pkg/cmd/experimental/syncgroups/cli/syncgroups.go]: what is this?

Comment (deads2k) [inline pkg/cmd/experimental/syncgroups/cli/syncgroups.go]: > what is this?

commented out due to pflags problems, but it allowed `--blacklist-group=group3`

File: pkg/cmd/experimental/syncgroups/cli/syncgroups.go (hunk 1 above)
Comment (deads2k) [inline pkg/cmd/experimental/syncgroups/cli/syncgroups.go]: > add to help text

done

**Comment by deads2k** [comment]
closing in favor of https://github.com/openshift/origin/pull/5108 which contains both.

**Pull request closed without merging**